### ISSUE TITLE
deploy and test integration with rdpc song 

### ISSUE THREAD

**Issue by blabadi** [icgc-argo/kship#3]
When rpdc song is deployed we need to test the kafka integration between rpdc and platform for song analysis events

# Expected Behaviour at the end of this ticket  
can publish a song analysis, and see the reflected changes in the platform

**Comment by rosibaj** [comment]
@blabadi i believe that this ticket is no longer blocked:
- production song and kafka on rdpc are deployed
- @rtisma is updating the kakfta queue name currently.

Is there anything else that was blocking this?

**Comment by blabadi** [comment]
Yeah I believe this can be unblocked now, I think the only missing thing is to register kship in ego as application which can be done as part of this ticket (the client Id and secret need to be obtained from kship by someone who has access to the rdpc cluster secrets)

**Comment by blabadi** [comment]
after some invistigation :

1- kship is not up in rdpc, timeouts connecting to kafka, probably needs proper network policies
2- kship health endpoint should be using /actuator/health (requires chart upgrade and redeploy) because it says healthy when it's not.

**Comment by rosibaj** [comment]
blocked: 

- work with @henro001 to verify network policies are set up correctly on kship
- update the kship health endpoint to actually reflect health (it just says true right now),  then the chart needs to be updated to reflect this

**Comment by blabadi** [comment]
blocked by collab data center problems

**Comment by blabadi** [comment]
messages now reach platform kafka
![image.png](https://images.zenhubusercontent.com/5c99083bd2fde800015cf9fa/2c2afc7a-5f1a-4371-b94a-ce0f024834eb)

problems found:
1- kship didn't have access to rdpc kafka: https://github.com/OICR/argo-infra/pull/342/files  (net policy)
2- gateway didn't have access to the rest proxy (net policy)

kafka ui didn't have net policy too (kafka ui is to inspect a topic)

**Comment by blabadi** [comment]
need to apply the changes into QA and prepare staging and prod PRs

**Comment by blabadi** [comment]
rdpc qa to platform qa now working

**Comment by rosibaj** [comment]
Cool will test some publishing today!

**Comment by blabadi** [comment]
rpdc prod to staging is blocked now due to network connectivity issues

**Comment by rosibaj** [comment]
Tested on QA: 
- submitted a tumour `sequencing_experiment` payload and published it under analysis-id `410b95dc-6d1f-4742-8b95-dc6d1f5742fa`
- checked the platform QA - dont see the expected analysis as a file: 
![image](https://user-images.githubusercontent.com/15824252/92524518-3d3bdd80-f1f0-11ea-8fd7-9cfccc7efc88.png)

Update kship did not seem to get message: 
```
2020-08-28 | 16:16:07.817 | 1 | KafkaConsumerDestination{consumerDestinationName='song_analysis', partitions=1, dlqName='kship-song-01-dlq'}.container-0-C-1 | ERROR | org.icgcargo.kship.ConsumerToHttp | failed to send message
org.springframework.web.reactive.function.client.WebClientResponseException$Forbidden: 403 Cannot get token, expected 2xx HTTP Status code 
```

**Comment by blabadi** [comment]
That's and old message 28th of August I don't think that is the reason

**Comment by blabadi** [comment]
I can actually see the message in kafka ui in qa, so it looks like the problem may be in maestro or the index :

![image.png](https://images.zenhubusercontent.com/5c99083bd2fde800015cf9fa/318ff40c-39c3-461e-b77e-1c6d6c4e7a4f)

**Comment by blabadi** [comment]
anyway from kship perspective everything looks good to me in QA. it kshipped !! ... I'll see my self out

**Comment by blabadi** [comment]
@yalturmes confirmed with IT that there is no rule to prevent connectivity in F5 and cumulus

**Comment by blabadi** [comment]
maestro in qa was pointing to rdpc prod song, which lead to not finding the analyses in the kafka messages, and that's why it didn't index anything so far

**Comment by blabadi** [comment]
@rosibaj wana give this another shot, try to unpublish and publish the same analyses you were trying before : 

21	0		a07fe97f-80e1-4e69-bfe9-7f80e18e6969	DASH-CA	PUBLISHED	song.collab
34	0		410b95dc-6d1f-4742-8b95-dc6d1f5742fa	DASH-CA	PUBLISHED	song.collab

**Comment by rosibaj** [comment]
@blabadi it worked
![image](https://user-images.githubusercontent.com/15824252/93128259-3522e880-f69d-11ea-9122-502e0d6b0f4f.png)

**Comment by blabadi** [comment]
regarding rdpc prod => staging I did further investigation and confirmed that there is a network configuration issue not application level that returns 403, @henro001  mentioned it could be the ingress access list. tbc

**Comment by blabadi** [comment]
examples a pod in rdpc prod cannot access  any ingress url in platform staging: 
```
srv $ wget https://platform.staging.icgc-argo.org/
Connecting to platform.staging.icgc-argo.org (206.108.121.128:443)
wget: server returned error: HTTP/1.1 403 Forbidden

/srv $ wget https://clinical.qa.argo.cancercollaboratory.org/api-docs/
Connecting to clinical.qa.argo.cancercollaboratory.org (142.1.177.161:443)
index.html           100% |******************************************************************|  3113  0:00:00 ETA
/srv $ rm index.html 
/srv $ wget https://clinical.platform.staging.icgc-argo.org/api-docs/
Connecting to clinical.platform.staging.icgc-argo.org (206.108.121.128:443)
wget: server returned error: HTTP/1.1 403 Forbidden
```

**Comment by henro001** [comment]
Updated private ingress configmap to add the public IP address of RDPC prod router to the whitelist 142.1.177.130
https://github.com/OICR/argo-infra/commit/cf06eed4275d82f8756a0c11f4d7111be3285424

**Comment by blabadi** [comment]
I confirmed that now kship is working end to end from rdpc prod to platform file centric index by checking maestro logs and the file repository, one analysis for examlpe was published today: 1a2bd6d8-be37-478f-abd6-d8be37578f29

**Comment by blabadi** [comment]
no further action on platform end in prod env for now
